### PR TITLE
Console device command list now working.

### PR DESCRIPTION
**Pull request by Slion**
Using DataContract to internalize config.
Adding ControlGroup and Function objects.

### CODE REVIEW

File: HarmonyHub/Entities/Device.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class Device : IComparable<Device>
     {
         [DataMember(Name = "Transport")]
         public int Transport { get; set; }
-        [DataMember(Name = "deviceTypeDisplayName")]
+        [DataMember(Name = "suggestedDisplay")]
```
Comment (Lakritzator) [inline HarmonyHub/Entities/Device.cs:15]: Damn copy / paste

File: HarmonyHub/Entities/Activity.cs [hunk 2]
```diff
@@ -19,7 +20,7 @@ public class Activity : IComparable<Activity>
         [DataMember(Name = "activityTypeDisplayName")]
         public string ActivityTypeDisplayName { get; set; }
         [DataMember(Name = "controlGroup")]
-        public IList<object> ControlGroup { get; set; }
+        public IList<object> ControlGroups { get; set; }
```
Comment (Lakritzator) [inline HarmonyHub/Entities/Activity.cs:23]: You added ControlGroup as a type, I guess it can be used here instead of object too?

**Comment by Lakritzator** [comment]
Good changes

P.S.
I would also be okay to add Json.net as the JSON parser,..

File: HarmonyHub/Entities/Activity.cs (hunk 2 above)
Comment (Slion) [inline HarmonyHub/Entities/Activity.cs:23]: Oops, missed that one. Was focused on the devices and forgot to check the activities.

Comment (Lakritzator) [inline HarmonyHub/Entities/Activity.cs:23]: No problem, that why it's nice to work together on a project... I miss something you spot, and the other way around!

I never just merge pull-request, I try to understand and check them.

**Pull request merged**